### PR TITLE
Added code file for review.

### PR DESCRIPTION
**Pull request by zwescoe**

### CODE REVIEW

File: CodeReview/ZakCode4Review.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            "0930", "1031", "1130", "1231"]
+
+# Create empty DataFrame to hold the data
+data = pd.DataFrame()
+# Create empty list to hold the file names
+filenames = []
+
+
+def get_data_year(station="9444900", begin="0101", end="0131", year="2014"):
+    """
```
Comment (mguerrap) [inline CodeReview/ZakCode4Review.py:28]: I think the function definition should not have the inputs defined. Like this:
def get_data_year(station, begin, end, year):

File: CodeReview/ZakCode4Review.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    ------------
+    station : string, default = "9444900" Port Townsend station
+        The tide station ID that the data is coming from
+
+    begin : string, default = "20140101", January 1st, 2014
+        The start date of the data set
+
+    end : string, default = "20140131", January 31st, 2014
+    The end date of the data set
+    """
```
Comment (mguerrap) [inline CodeReview/ZakCode4Review.py:42]: Year is missing in the inputs explanations

File: CodeReview/ZakCode4Review.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+           "level&application=NOS.COOPS.TAC.WL&station=" + station + "&" +
+           "begin_date=" + first_index + "&end_date=" + second_index +
+           "&datum=MLLW&units=" + "english&time_zone=GMT&format=csv")
+    outfile = station + first_index + ".csv"
+    req = requests.get(URL)
+    assert req.status_code == 200
+    with open(outfile, "wb") as f:
+        f.write(req.content)
+    filenames.append(outfile)
+
```
Comment (mguerrap) [inline CodeReview/ZakCode4Review.py:56]: Is here the end of the function? Maybe the function can be moved to an external .py file

File: CodeReview/ZakCode4Review.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+# Create empty DataFrame to hold the data
+data = pd.DataFrame()
+# Create empty list to hold the file names
+filenames = []
+
+
+def get_data_year(station="9444900", begin="0101", end="0131", year="2014"):
+    """
+    This fuction gets the data from 2014 in month chunks.
```
Comment (mguerrap) [inline CodeReview/ZakCode4Review.py:29]: If year is an input, then this would be for any year?

File: CodeReview/ZakCode4Review.py (hunk 3 above)
Comment (mguerrap) [inline CodeReview/ZakCode4Review.py:56]: Does the function writes the monthly files in the computer? I got confused

**Review by mguerrap — COMMENTED** [review]
Hey Zak, here are my comments.

File: CodeReview/ZakCode4Review.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    # For each station set of data, read the files and convert to one file
+    for i in range(0, len(filenames)):
+        buff = pd.read_csv(filenames[i])
+        # Drop the unnecessary columns from the data
+        buff.drop(buff.columns[[3, 4, 5, 6, 7]], axis=1, inplace=True)
+        # Trim off the white space on the column names
+        buff.columns = ["Date Time", "Water Level", "Sigma"]
+        data = data.append(buff)
+    # Sets outfile to the station name and year
+    outfile = (year + "_" + station_id_to_names[filenames[0][0:7]] + ".csv")
```
Comment (reedan88) [inline CodeReview/ZakCode4Review.py:75]: Python has lots of different ways to handle splitting and concatenating strings. I'd try looking at split, tok, and slice methods over counting indices. Makes for a more "flexible" code.

File: CodeReview/ZakCode4Review.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    outfile = station + first_index + ".csv"
+    req = requests.get(URL)
+    assert req.status_code == 200
+    with open(outfile, "wb") as f:
+        f.write(req.content)
+    filenames.append(outfile)
+
+
+# Prompts the user for the year of data they want
+year = str(input("What year of data do you want? "))
```
Comment (reedan88) [inline CodeReview/ZakCode4Review.py:59]: What happens if I enter a year that doesn't exists? I might put in a check here or an interactive widget that constrains the choices.

File: CodeReview/ZakCode4Review.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    # Concatinate the year to the start and end dates
+    first_index = year + begin
+    second_index = year + end
+    URL = ("https://tidesandcurrents.noaa.gov/api/datagetter?product=water_" +
+           "level&application=NOS.COOPS.TAC.WL&station=" + station + "&" +
+           "begin_date=" + first_index + "&end_date=" + second_index +
+           "&datum=MLLW&units=" + "english&time_zone=GMT&format=csv")
+    outfile = station + first_index + ".csv"
+    req = requests.get(URL)
+    assert req.status_code == 200
```
Comment (reedan88) [inline CodeReview/ZakCode4Review.py:52]: I'd put a raise exception in here to catch if the assertion fails, you can retain the error and know that the URL has gone bad.

File: CodeReview/ZakCode4Review.py [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    with open(outfile, "wb") as f:
+        f.write(req.content)
+    filenames.append(outfile)
+
+
+# Prompts the user for the year of data they want
+year = str(input("What year of data do you want? "))
+# Collects the data for the whole year saved in monthly csv files
+for station in station_names_to_id:
```
Comment (reedan88) [inline CodeReview/ZakCode4Review.py:61]: Is there a way these for loops could be repackaged into a function, so the processing can take place in the background and make the script cleaner?

**Pull request merged**